### PR TITLE
Adds common test for navstar overlay

### PR DESCRIPTION
**Pull request by GoelDeepak**
Summary
-- cowboy exposes overlay-agent endpoint
-- navstar_overlay_poller pulls it and pushes to lashup
-- navstar_overlay_lashup_kv reads and parses it
-- test verifies the parse result
-- run_command for test is a print statement

### CODE REVIEW

**Comment by GoelDeepak** [comment]
@sargun  Overall code coverage from 33% to 54%

Here are the individual ones:
1. navstar_overlay_helper - 100%
2. navstar_overlay_sup - 100%
3. navstar_overlay_configure - 97%
4. navstar_overlay_lashup_kv_listener - 86%
5. navstar_overlay_poller - 76%
6. navstar_overlay_app - 44%

File: apps/navstar_overlay/test/navstar_overlay_SUITE.erl [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+maybe_kill(Node) ->
+  case ci() of
+    true ->
+      Command = io_lib:format("pkill -9 -f ~s", [Node]),
+      os:cmd(Command);
+    false ->
+      ok
+  end.
+
+%% Borrowed from the ct_slave module
```
Comment (sargun) [inline apps/navstar_overlay/test/navstar_overlay_SUITE.erl:170]: Why do we have to do this?

Comment (GoelDeepak) [reply apps/navstar_overlay/test/navstar_overlay_SUITE.erl:170]: Copy & paste. I thought this is a safe was of shutting down a node. No?

File: apps/navstar_overlay/src/navstar_overlay_configure.erl [hunk 2]
```diff
@@ -90,6 +90,10 @@ configure_overlay_entry(Overlay, _VTEPIPPrefix = {VTEPIP, _PrefixLen}, LashupVal
     FormattedVTEPIP = inet:ntoa(VTEPIP),
     FormattedSubnetIP = inet:ntoa(SubnetIP),
 
+    maybe_print_parameters([FormattedAgentIP, binary_to_list(VTEPName),
```
Comment (sargun) [inline apps/navstar_overlay/src/navstar_overlay_configure.erl:94]: Maybe add a comment here indicating that this is a testing fixture

File: apps/navstar_overlay/src/navstar_overlay_helper.erl [hunk 3]
```diff
@@ -17,7 +17,7 @@ run_command(Command, Opts) ->
 
 -spec(run_command(Command :: string()) ->
     {ok, Output :: string()} | {error, ErrorCode :: non_neg_integer(), ErrorString :: string()}).
--ifdef(DEV).
+-ifdef(TEST).
```
Comment (sargun) [inline apps/navstar_overlay/src/navstar_overlay_helper.erl]: we want to make sure that these don't during both testing and dev.

**Review by sargun — CHANGES_REQUESTED** [review]

**Comment by GoelDeepak** [comment]
@sargun I have incorporated your comments. Do you have some more or this is fine to merge?

**Review by sargun — APPROVED** [review]

**Pull request merged**